File: tps/tp1/palindromo.java
  import java.util.Scanner;
  

     public class palindromo{
        Scanner sc = new Scanner(System.in);

        public static void main(String[] args)
        {
            while (getPalavra() != "FIM") {
                
            
            if (ehPalindromo(getPalavra()))
            {
                System.out.printf("SIM\n");
            }
            else
            System.out.printf("NAO\n");
        }
        }

        String getPalavra()
        {
            String palavra = sc.nextLine();
            return palavra;
        }


        boolean ehPalindromo(String palavra)
        {
            
            int tamanho = palavra.length() - 1;
            
            for(int i =0; i < tamanho/2 ; i++)
            {
                if (palavra.charAt(0 ) != palavra.charAt(tamanho - i))
                {
                    return false; // Não é um palíndromo
                }
            }
        
            return true;
        }



     }

File: tps/tp1/palindromo.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <stdbool.h>

void getstring(char palavra[], int maxSize);
bool ehPalindromo(char palavra[]);

int main()
{
    char palavra[50];
    getstring(palavra, sizeof(palavra));

    if(ehPalindromo(palavra))
    {
        printf("SIM\n");
    }
    else
    {
        printf("NAO\n");
    }

    return 0;
}

// Gera uma string a partir da entrada do teclado.
void getstring(char palavra[], int maxSize)
{
    fgets(palavra, maxSize, stdin);

    //Remove espaço em branco da string para nao ocorrer erros na comparação
    size_t tamanho = strlen(palavra);
    if (tamanho > 0 && palavra[tamanho - 1] == '\n') {
        palavra[tamanho - 1] = '\0';
    }
}


//Confere se eh palindromo ou nao e retorna um bool dependendo do caso.
bool ehPalindromo(char palavra[])
{
    int tamanho = strlen(palavra);

    for(int i = 0; i < tamanho / 2; i++) 
    {
        if(palavra[i] != palavra[tamanho - i - 1]) 
        {
            return false; 
        }
    }
    return true; 
}
